vcf_summary: Fix basic checklist items for DX apps

msimbirsky opened the pull request:
- dxapp.json: Add input help text, citation stub, regions, timeout, distro
- Note an unimplemented option
- Consistent whitespace/indentation in the main script
- Remove dead code
- Remove boilerplate comments from Readme.md and main script
- Set shell options '-e -o pipefail'

etal committed:
vcf_summary: Fix basic checklist items for DX apps

- dxapp.json: Add input help text, citation stub, regions, timeout, distro
- Note an unimplemented option
- Consistent whitespace/indentation in the main script
- Remove dead code
- Remove boilerplate comments from Readme.md and main script
- Set shell options '-e -o pipefail'

msimbirsky commented:
#dxapp.json
- [ ] Upgrade to 14.04
- [ ] Add help and extension to input spec.
- [ ] In `details` field specify: `whatsNew`and `citation` (leave blank for citation.
- [ ] In `details` consider specifying `upstreamAuthor` and `upstreamVersion` fields when applicable.
- [ ] Use `regionalOptions` 
- [ ] Set applicable timeout (Be relax) and restart policy in [`runSpec`](https://wiki.dnanexus.com/API-Specification-v1.0.0/IO-and-Run-Specifications).
- [ ] Add subjob restartability when applicable.
- [ ] `version` field, set proper [semantic versioning](https://semver.org/).
- [ ] Setting `"openSource": true`. 
- [ ] Run [dxapp linter](https://github.com/dnanexus/RoughER/blob/master/scripts/dxapp_lint.py) [[permalink](https://github.com/dnanexus/RoughER/blob/ecb3ea3f97870c020f93bf4723afc9f40a310e33/scripts/dxapp_lint.py)] on `dxapp.json`.
- [ ] Use AssetDepend instead of execDepend
- [ ] Add default instance type that works for a general use case.
- [ ] add new input if applicable

#src/code.sh src/code.py
- [ ] GATK tarball should be removed from the resource project (or make a new resource project), and the applet should require GATK tarball as a required input. This way, the users are enforced to provide their own tarball. This would avoid trouble from non-academic entities without GATK license to use your tarball.

# Readme.md
- [ ] Improve Readme file to contain applet function, use case, input/output description, underline execution if it’s a wrapper.
- [ ] Read documentation for sanity check. (Unpublished app URL will be: https://platform.dnanexus.com/app/app-xxxx)

# Test
- [ ] Basic testing using given input in Box or expected input
- [ ] Make a Pull Request to `master` in https://github.com/RitchieLab/dnanexus

# Production
- [ ] Co-test with Ritchie group
- [ ] Help Ritchie group to publish (Give dx command)
- [ ] Verify the versions that appear on the platform are the intended versions
- [ ] Set up demo workflow in public project.

msimbirsky commented:
Missing "variant-level summary" script

slzarate commented on sequencing/vcf_summary/dxapp.json:
```diff
@@ -3,34 +3,42 @@
+  },
   "inputSpec": [
     {
       "name": "vcf_fn",
```
Change to `variants_vcfgzs`

slzarate commented on sequencing/vcf_summary/dxapp.json:
```diff
@@ -3,34 +3,42 @@
     {
       "name": "vcf_fn",
-      "label": "VCF File",
+      "label": "VCF file",
```
I think we decided on capitalizing all words in the label

slzarate commented on sequencing/vcf_summary/Readme.md:
```diff
@@ -1,21 +1,6 @@
 # VCF Summary Statistics (DNAnexus Platform App)
 
-Generate summary statistics for a VCF file (by sample and by variant)
+Generate summary statistics for a VCF file (by sample and by variant).
```
to write

Commit:
Update version number

etal commented on sequencing/vcf_summary/dxapp.json:
```diff
@@ -3,34 +3,42 @@
   "summary": "Generate summary statistics for a VCF file (by sample and by variant)",
   "dxapi": "1.0.0",
-  "version": "0.0.1",
+  "version": "0.1.0",
```
I think we're setting the versions to 1.0.0, not 0.1.0. (I had it wrong initially.)

Commit:
Update app

- input file nomenclature
- remove extraneous 'dx describe' commands
- handle resource project correctly
- update resource project ids
- remove empty output for unimplemented option
- replace execDepends with assetDepends
- update readme.md

Commit:
Update biofilter folder path and Loki DB path

Commit:
Remove option to run variants summary and always run sample summary

Commit:
Cosmetic fixes

msimbirsky commented:
@etal @slzarate Review amendments are done
